grails-app/services/de/iteratec/osm/measurement/environment/wptserver/ResultPersisterService.groovy

```diff
@@ -94,20 +95,19 @@ class ResultPersisterService implements iResultListener {
     }
 
     @Transactional(propagation = Propagation.REQUIRES_NEW)
-    void checkJobAndLocation(WptResultXml resultXml, WebPageTestServer wptserverOfResult) throws OsmResultPersistanceException {
+    void checkJobAndLocation(WptResultXml resultXml, WebPageTestServer wptserverOfResult, long jobId) throws OsmResultPersistanceException {
         Job job
         performanceLoggingService.logExecutionTime(DEBUG, "get or persist Job ${resultXml.getLabel()} while processing test ${resultXml.getTestId()}...", 4) {
-            String jobLabel = resultXml.getLabel()
-            job = jobDaoService.getJob(jobLabel)
-            if (job == null) throw new OsmResultPersistanceException("No measurement job could be found for label from result xml: ${jobLabel}")
+            job = jobDaoService.getJob(jobId)
```

Comment:
Getting the job and checking for it to be non-null is done in all of the `@Transactional` methods. Can you extract the logic? Is `getJob` called from a location that actually expects it to return null in case the job does not exist (seems strange when querying by id)? Otherwise that method could check this directly.